_posts/2023-05-02-12.0.0-release.md

```diff
@@ -0,0 +1,275 @@
+* A new cgo-based Allocator that does not depend on libarrow has been added to the memory package ([GH-33901](https://github.com/apache/arrow/issues/33901))
+* CSV Reader and Writer now support Extension type arrays ([GH-34334](https://github.com/apache/arrow/issues/34334))
+* Fixed bug preventing the reading of IPC streams/files with compression enabled but uncompressed buffers ([GH-34385](https://github.com/apache/arrow/issues/34385))
+* Added interface which can be added to an `ExtensionType` to allow Builders to be created via `NewBuilder`, enabling easy building of nested fields containing extension types ([GH-34453](https://github.com/apache/arrow/issues/34453))
+* Added utilities to perform Array diffing ([GH-34790](https://github.com/apache/arrow/issues/34790))
+* Added `SetColumn` method to `arrow.Record` ([GH-34832](https://github.com/apache/arrow/issues/34832))
+* Added `GetValue` method to `arrow.Metadata` ([GH-34855](https://github.com/apache/arrow/issues/34855))
+* Added `Pow` method to `decimal128.Num` and `decimal256.Num` ([GH-34863](https://github.com/apache/arrow/issues/34863))
+
+#### Flight
+
+* Fixed bug in `StreamChunks` for Flight SQL to correctly propagate to the gRPC client ([GH-33717](https://github.com/apache/arrow/issues/33717))
+* Fixed issue that prevented compatibility with gRPC < v1.45 ([GH-33734](https://github.com/apache/arrow/issues/33734))
+* Added support to bind a `RecordReader` for supplying parameters to a Flight SQL Prepared statement ([GH-33794](https://github.com/apache/arrow/issues/33794))
+* Prepared Statement methods for FlightSQL client now allows gRPC Call Options ([GH-33867](https://github.com/apache/arrow/issues/33867))
+* FlightSQL Extensions have been implemented (for transactions and Substrait support) ([GH-33935](https://github.com/apache/arrow/issues/33935))
+* A driver compatible with `database/sql` for FlightSQL has been added ([GH-34332](https://github.com/apache/arrow/issues/34332))
+
+#### Compute
+
+* "run_end_encode" and "run_end_decode" functions added to compute functions ([GH-20408](https://github.com/apache/arrow/issues/20408))
+* "unique" function added ([GH-34171](https://github.com/apache/arrow/issues/34171))
+
+### Parquet
+
+* `pqarrow` pkg now handles DICTIONARY fields natively ([GH-33466](https://github.com/apache/arrow/issues/33466))
+* Fixed rare panic when writing list of 8 structs ([GH-33600](https://github.com/apache/arrow/issues/33600))
+* Added support for `pqarrow` pkg to write LargeString and LargeBinary types ([GH-33875](https://github.com/apache/arrow/issues/33875))
+* Fixed bug where `pqarrow.NewSchemaManifest` created the wrong field type for Array Object fields ([GH-34101](https://github.com/apache/arrow/issues/34101))
+* Added support to Parquet Writer for Extension type Arrays ([GH-34330](https://github.com/apache/arrow/issues/34330))
+
+## Java notes
```

Comment:
```suggestion
## Java notes

* Update Java JNI modules to consider Arrow ACERO [GH-34862](https://github.com/apache/arrow/issues/34862)
* Ability to register additional GRPC services with FlightServer [GH-34778](https://github.com/apache/arrow/issues/34778)
* Allow sending custom headers/properties through Arrow Flight SQL JDBC [GH-33874](https://github.com/apache/arrow/issues/33874)
```

@davisusanibar @lidavidm this is the list of Java issues for 12.0.0: https://github.com/apache/arrow/issues?q=is%3Aissue+label%3A%22Component%3A+Java%22+is%3Aclosed+milestone%3A12.0.0+
I added some notes but is there anything else you would like to add?